include/SFML/System/Vector2.hpp

```diff
@@ -68,14 +73,132 @@ class Vector2
+    /// The angle determines how much you have to rotate \c *this
+    ///  until it points to the same direction as \c rhs.
+    /// \pre Neither \c *this nor \c rhs is a zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle signedAngleTo(const Vector2& rhs) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Angle from +X or (1,0) vector <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds to 0 degrees, (0,1) corresponds to 90 degrees.
+    /// 
+    /// \return Angle in degrees in the interval [-180,180].
+    /// \pre \c vector is no zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle polarAngle() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Rotate by given angle <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds 0 degrees, (0,1) corresponds 90 degrees.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 rotatedBy(Angle angle) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Projection of \c vector onto \c axis <i><b>(floating-point)</b></i>.
+    /// 
+    /// \param vector Vector to project.
+    /// \param axis Vector being projected onto. Need not be normalized, but must not have length zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 projectedOnto(const Vector2& axis) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Returns a perpendicular vector.
+    /// 
+    /// Returns \c vector rotated by 90 degrees counter clockwise; (x,y) becomes (-y,x).
+    /// For example, the vector (1,0) is transformed to (0,1).
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2 perpendicular() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Dot product of two 2D vectors.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr T dot(const Vector2& rhs) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Z component of the cross product of two 2D vectors.
+    /// 
+    /// Treats the operands as 3D vectors, computes their cross product and returns the result's Z component
+    ///  (X and Y components are always zero).
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr T cross(const Vector2& rhs) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Component-wise multiplication of \c *this and \c rhs.
+    /// 
+    /// Computes <tt>(lhs.x*rhs.x, lhs.y*rhs.y)</tt>. Main use case are scales.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Vector2 cwiseMul(const Vector2& rhs) const;
```

Comment:
I don't mind this name, but maybe consider `compWiseMul`?

Reply:
This is even more verbose, while not really being much more expressive. I'd rather consider overloading `*` and `/` directly, like it's done in every other vector library 🙂